### ISSUE TITLE
Complex Valued Neural Network

### ISSUE THREAD

**Issue by Vlodson** [ftn-ai-lab/nm-2023#5]
### Tim
Vladimir Lunić E2-3-2023

### Definicija problema
Implementirati modul za duboke neuronske mreže koje koriste isključivo vrednosti iz kompleksnog domena. Implementirani modul testirati na sintetičkim podacima spram obične duboke neuronske mreže, kao i na skupu podataka MSTAR-Public na public-chip podskupu.

### Skupovi podataka
Sintetički podaci će biti napravljeni tako da bi se mogle porediti metrike, efikasnosti i razlike između CVNN i DNN. Primeri skupova su binarna klasifikacija dva linearno razdvojiva gausovski generisanim klastera, višeklasna klasifikacija na gausovski generisanim klasterima i sl.

MSTAR-Public skup podataka je skup slika sa SAR vojnih radara za praćenje objekata na zemlji. Izlaz iz SAR je slika objekta sa zemlje u kompleksnom domenu koja se najčešće razdvaja na magnitudu i fazu kako bi bila čitljiva za ljude.

### Metodologija
Implementirati modul koji će sadržati apstrakciju Dense sloja na kompleksne vrednosti, nove aktivacione funkcije za kompleksni domen, funkcije greške za kompleksni domen, novi način klasifikacije vektora u kompleksnom domenu, metrike za testiranje i treniranje, apstrakciju Sequential modela nalik na keras.models.Sequential, apstrakcija History nalik na tensorflow.callbacks.History, CPU optimizacije pomoću Numba, GPU optimizacije pomoću CuPy.

### Evaluacija
Porediti Accuracy, Precision, Recall i F1 za DNN i CVNN na sintetičkim podacima. Poređenje brzine pre i posle optimizacija nad CVNN. Poređenje rezultata na MSTAR skupu sa rezultatima dobijenim u SIAP delu projekta.

**Comment by vdragan1993** [comment]
Tema odobrena.
Ostavite link ka GitHub repozitorijumu projekta.
Srećan rad.

**Comment by Vlodson** [comment]
[GitHub repozitorijum](https://github.com/Vlodson/Complex-Valued-Neural-Network)
